src/web/page/templates.rs

```diff
@@ -0,0 +1,191 @@
+    tera.register_filter("timeformat", timeformat);
+    tera.register_filter("dbg", dbg);
+    tera.register_filter("dedent", dedent);
+
+    Ok(tera)
+}
+
+/// Returns an `Option<GlobalAlert>` in json form for templates
+fn global_alert(args: &HashMap<String, Value>) -> TeraResult<Value> {
+    debug_assert!(args.is_empty(), "global_alert takes no args");
+
+    Ok(TEMPLATE_DATA.global_alert.clone())
+}
+
+/// Returns the version of docs.rs, takes the `safe` parameter which can be `true` to get a url-safe version
+fn docsrs_version(args: &HashMap<String, Value>) -> TeraResult<Value> {
+    debug_assert!(
+        args.is_empty(),
+        "docsrs_version only takes no args, to get a safe version use `docsrs_version() | slugify`",
+    );
+
+    Ok(TEMPLATE_DATA.docsrs_version.clone())
+}
+
+/// Returns the current rustc resource suffix
+fn rustc_resource_suffix(args: &HashMap<String, Value>) -> TeraResult<Value> {
+    debug_assert!(args.is_empty(), "rustc_resource_suffix takes no args");
+
+    Ok(TEMPLATE_DATA.resource_suffix.clone())
+}
+
+/// Prettily format a timestamp
+// TODO: This can be replaced by chrono
+fn timeformat(value: &Value, args: &HashMap<String, Value>) -> TeraResult<Value> {
+    let fmt = if let Some(Value::Bool(true)) = args.get("relative") {
+        let value = time::strptime(value.as_str().unwrap(), "%Y-%m-%dT%H:%M:%S%z").unwrap();
+
+        super::super::duration_to_str(value.to_timespec())
+    } else {
+        const TIMES: &[&str] = &["seconds", "minutes", "hours"];
+
+        let mut value = value.as_f64().unwrap();
+        let mut chosen_time = &TIMES[0];
+
+        for time in &TIMES[1..] {
+            if value / 60.0 >= 1.0 {
+                chosen_time = time;
+                value /= 60.0;
+            } else {
+                break;
+            }
+        }
+
+        // TODO: This formatting section can be optimized, two string allocations aren't needed
+        let mut value = format!("{:.1}", value);
+        if value.ends_with(".0") {
+            value.truncate(value.len() - 2);
+        }
+
+        format!("{} {}", value, chosen_time)
+    };
+
+    Ok(Value::String(fmt))
+}
+
+/// Print a tera value to stdout
+fn dbg(value: &Value, _args: &HashMap<String, Value>) -> TeraResult<Value> {
+    println!("{:?}", value);
+
+    Ok(value.clone())
+}
+
+/// Dedent a string by removing all leading whitespace
+fn dedent(value: &Value, _args: &HashMap<String, Value>) -> TeraResult<Value> {
+    let string = value.as_str().expect("dedent takes a string");
+
+    Ok(Value::String(
+        string
+            .lines()
+            .map(|l| l.trim_start())
+            .collect::<Vec<&str>>()
+            .join("\n"),
+    ))
+}
+
+#[cfg(test)]
+mod tests {
+    // TODO: This will fail as long as there are `.hbs` files in `templates/`
+    // #[test]
+    // fn test_templates_are_valid() {
+    //     let tera = load_templates().unwrap();
+    //     tera.check_macro_files().unwrap();
+    // }
```

Comment:
An approach here could be to temporarily store the templates in `templates-tera/`, which we'd move to `templates/` after the whole migration is finished.